Dep bump to runc 1.1.0, cadvisor 0.44.0

ehashman opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind cleanup
/priority important-soon
/sig node
/cc @odinuge @dims @mrunalp 

#### What this PR does / why we need it:

Non-draft rebased follow-up to https://github.com/kubernetes/kubernetes/pull/107149

This updates vendored runc/libcontainer to 1.1.0,
and google/cadvisor to a version updated to runc 1.1.0
(https://github.com/google/cadvisor/pull/3048).

Changes in vendor are generated by (roughly):

        ./hack/pin-dependency.sh github.com/google/cadvisor v0.44.0
        ./hack/pin-dependency.sh github.com/opencontainers/runc v1.1.0
        ./hack/update-vendor.sh
        ./hack/lint-dependencies.sh # And follow all its recommendations.
        ./hack/update-vendor.sh
        ./hack/update-internal-modules.sh
        ./hack/lint-dependencies.sh # Re-check everything again.

The changes (mostly in pkg/kubelet/cm) are there to adopt changed
runc 1.1 API, and simplify things a bit. In particular:

1. simplify cgroup manager instantiation, using a new, easier way of
   libcontainers/cgroups/manager.New;

2. replace libcontainerAdapter with a boolean variable (all it did
   was passing on whether systemd manager should be used);

3. trivial change due to removed cgroupfs.HugePageSizes and added
    cgroups.HugePageSizes();

4. do not calculate cgroup paths in update / destroy, since libcontainer
   cgroup managers now calculate the paths upon creation (previously,
   they were doing that only in Apply, so using e.g. Set or Destroy right
   after creation was impossible without specifying paths).

We currently still calculate cgroup paths in Exists -- this is to be
addressed separately.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
none

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Update runc to 1.1.0
Update cadvisor to 0.44.0
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

kolyshkin committed:
pkg/kubelet: changes to update runc to 1.1.0

The changes (mostly in pkg/kubelet/cm) are there to adopt changed
runc 1.1 API, and simplify things a bit. In particular:

1. simplify cgroup manager instantiation, using a new, easier way of
   libcontainers/cgroups/manager.New;

2. replace libcontainerAdapter with a boolean variable (all it did
   was passing on whether systemd manager should be used);

3. trivial change due to removed cgroupfs.HugePageSizes and added
    cgroups.HugePageSizes();

4. do not calculate cgroup paths in update / destroy, since libcontainer
   cgroup managers now calculate the paths upon creation (previously,
   they were doing that only in Apply, so using e.g. Set or Destroy right
   after creation was impossible without specifying paths).

We currently still calculate cgroup paths in Exists -- this is to be
addressed separately.

Co-Authored-By: Elana Hashman <ehashman@redhat.com>

ehashman commented on go.mod:
```diff
@@ -354,7 +354,7 @@ replace (
 	github.com/russross/blackfriday => github.com/russross/blackfriday v1.5.2
 	github.com/russross/blackfriday/v2 => github.com/russross/blackfriday/v2 v2.1.0
-	github.com/seccomp/libseccomp-golang => github.com/seccomp/libseccomp-golang v0.9.1
+	github.com/seccomp/libseccomp-golang => github.com/seccomp/libseccomp-golang v0.9.2-0.20210429002308-3879420cc921
```
This one appears to be unavoidable... runc has a direct dep on it https://github.com/opencontainers/runc/blob/86d6898f3052acba1ebcf83aa2eae3f6cc5fb471/go.mod#L17

kolyshkin replied:
Yes, libseccomp-golang did not made a release since 2019, and the released version is no longer adequate (mostly due to seccomp-notify functionality which runc now supports/requires)

ehashman commented:
/test pull-kubernetes-node-kubelet-serial-containerd

ehashman commented:
/milestone v1.24

dims commented:
The dependency change LGTM. Please assign to Jordan when this is green as i don't have root OWNERS approval

thanks!

bobbypage commented:
/lgtm

bobbypage commented:
Thanks for rebasing this @ehashman and @kolyshkin for the update!

odinuge reviewed (approved):
Thanks! Looks good!

/lgtm

ehashman commented:
/triage accepted
/assign @mrunalp @dims 

for node/deps approval

ehashman commented:
> The dependency change LGTM. Please assign to Jordan when this is green as i don't have root OWNERS approval

oops

/unassign @dims
/assign @liggitt

liggitt commented:
mechanics look good, and we dropped a dep :)

```diff
{
 	"directDependencies": [16](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/109029/check-dependency-stats/1507428262307958784#1:build-log.txt%3A16)2,
-	"transitiveDependencies": 258,
-	"totalDependencies": 319,
+	"transitiveDependencies": 257,
+	"totalDependencies": 318,
 	"maxDepthOfDependencies": 29
 }
```

We have a dependency revert going in (https://github.com/kubernetes/kubernetes/pull/109040) because it looks like a grpc bump this morning may have been what turned test-integration red. Once that goes in, can tag this.

There's also https://github.com/kubernetes/kubernetes/pull/108988 and https://github.com/kubernetes/kubernetes/pull/108828 in flight, so at least one of these will likely need to rebase on whichever merges first

mrunalp commented:
Looks good :+1:

liggitt commented on vendor/github.com/opencontainers/runc/libcontainer/cgroups/utils.go:
```diff
@@ -49,6 +49,24 @@ func IsCgroup2UnifiedMode() bool {
+				isHybrid = false
+				return
+			}
+			panic(fmt.Sprintf("cannot statfs cgroup root: %s", err))
```
this is called from an init() func in vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/fs.go, so anything that links this in will panic if any error that IsNotExist is encountered on startup (like a permissions error)

doesn't that seem problematic?

ehashman replied:
From what I can tell from a git grep, the only components that import this are the Kubelet and a few test files. I think the risk is low, and we could fix this in a patch version backport bump of runc to follow?

<details>

```
ehashman@red-dot:~/src/k8s$ git grep 'github.com/opencontainers/runc/libcontainer/cgroups'
cmd/kubelet/app/server_linux.go:        libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/cm/cgroup_manager_linux.go: libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/cm/cgroup_manager_linux.go: cgroupfs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
pkg/kubelet/cm/cgroup_manager_linux.go: cgroupfs2 "github.com/opencontainers/runc/libcontainer/cgroups/fs2"
pkg/kubelet/cm/cgroup_manager_linux.go: "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
pkg/kubelet/cm/cgroup_manager_linux.go: cgroupsystemd "github.com/opencontainers/runc/libcontainer/cgroups/systemd"
pkg/kubelet/cm/container_manager_linux.go:      "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/cm/container_manager_linux.go:      cgroupfs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
pkg/kubelet/cm/container_manager_linux.go:      cgroupfs2 "github.com/opencontainers/runc/libcontainer/cgroups/fs2"
pkg/kubelet/cm/container_manager_linux_test.go: "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/cm/helpers_linux.go:        libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/cm/pod_container_manager_linux.go:  libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/cm/qos_container_manager_linux.go:  libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/cm/qos_container_manager_linux.go:  cgroupfs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
pkg/kubelet/cm/util/cgroups_linux.go:   libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/eviction/threshold_notifier_linux.go:       libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/kuberuntime/kuberuntime_container_linux.go: libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
pkg/kubelet/kuberuntime/kuberuntime_container_linux.go: cgroupfs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
pkg/kubelet/kuberuntime/kuberuntime_container_linux_test.go:    cgroupfs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
pkg/util/oom/oom_linux_test.go: "github.com/opencontainers/runc/libcontainer/cgroups"
test/e2e_node/resource_collector.go:    "github.com/opencontainers/runc/libcontainer/cgroups"
test/e2e_node/utils_linux.go:   libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/google/cadvisor/container/common/helpers.go:  "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/google/cadvisor/container/containerd/handler.go:      "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/google/cadvisor/container/crio/handler.go:    "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/google/cadvisor/container/libcontainer/handler.go:    "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/google/cadvisor/container/libcontainer/handler.go:    "github.com/opencontainers/runc/libcontainer/cgroups/fs2"
vendor/github.com/google/cadvisor/container/libcontainer/helpers.go:    "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/google/cadvisor/container/libcontainer/helpers.go:    fs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
vendor/github.com/google/cadvisor/container/libcontainer/helpers.go:    fs2 "github.com/opencontainers/runc/libcontainer/cgroups/fs2"
vendor/github.com/google/cadvisor/container/raw/handler.go:     "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/google/cadvisor/manager/manager.go:   "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/google/cadvisor/resctrl/utils.go:     "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/google/cadvisor/resctrl/utils.go:     "github.com/opencontainers/runc/libcontainer/cgroups/fs2"
vendor/github.com/google/cadvisor/utils/sysfs/sysfs.go:// isCPUOnline is copied from github.com/opencontainers/runc/libcontainer/cgroups/fs and modified to suite cAdvisor
vendor/github.com/opencontainers/runc/libcontainer/cgroups/ebpf/devicefilter/devicefilter.go:   devicesemulator "github.com/opencontainers/runc/libcontainer/cgroups/devices"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/blkio.go: "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/cpu.go:   "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/cpu.go:   "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/cpuacct.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/cpuacct.go:       "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/cpuset.go:        "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/cpuset.go:        "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/devices.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/devices.go:       cgroupdevices "github.com/opencontainers/runc/libcontainer/cgroups/devices"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/freezer.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/fs.go:    "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/fs.go:    "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/hugetlb.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/hugetlb.go:       "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/memory.go:        "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/memory.go:        "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/name.go:  "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/net_cls.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/net_prio.go:      "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/perf_event.go:    "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/pids.go:  "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/pids.go:  "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/cpu.go:  "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/cpu.go:  "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/cpuset.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/create.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/devices.go:      "github.com/opencontainers/runc/libcontainer/cgroups/ebpf"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/devices.go:      "github.com/opencontainers/runc/libcontainer/cgroups/ebpf/devicefilter"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/freezer.go:      "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/fs2.go:  "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/fs2.go:  "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/hugetlb.go:      "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/hugetlb.go:      "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/io.go:   "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/memory.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/memory.go:       "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/pids.go: "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/pids.go: "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/fscommon/utils.go:   "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/systemd/common.go:   cgroupdevices "github.com/opencontainers/runc/libcontainer/cgroups/devices"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/systemd/unsupported.go:      "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/systemd/v1.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/systemd/v1.go:       "github.com/opencontainers/runc/libcontainer/cgroups/fs"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/systemd/v2.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/cgroups/systemd/v2.go:       "github.com/opencontainers/runc/libcontainer/cgroups/fs2"
vendor/github.com/opencontainers/runc/libcontainer/configs/validate/validator.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/container_linux.go:  "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/factory_linux.go:    "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/factory_linux.go:    "github.com/opencontainers/runc/libcontainer/cgroups/fs"
vendor/github.com/opencontainers/runc/libcontainer/factory_linux.go:    "github.com/opencontainers/runc/libcontainer/cgroups/fs2"
vendor/github.com/opencontainers/runc/libcontainer/factory_linux.go:    "github.com/opencontainers/runc/libcontainer/cgroups/systemd"
vendor/github.com/opencontainers/runc/libcontainer/init_linux.go:       "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/notify_linux_v2.go:  "github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
vendor/github.com/opencontainers/runc/libcontainer/process_linux.go:    "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/process_linux.go:    "github.com/opencontainers/runc/libcontainer/cgroups/fs2"
vendor/github.com/opencontainers/runc/libcontainer/rootfs_linux.go:     "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/github.com/opencontainers/runc/libcontainer/rootfs_linux.go:     "github.com/opencontainers/runc/libcontainer/cgroups/fs2"
vendor/github.com/opencontainers/runc/libcontainer/stats_linux.go:      "github.com/opencontainers/runc/libcontainer/cgroups"
vendor/modules.txt:github.com/opencontainers/runc/libcontainer/cgroups
vendor/modules.txt:github.com/opencontainers/runc/libcontainer/cgroups/devices
vendor/modules.txt:github.com/opencontainers/runc/libcontainer/cgroups/ebpf
vendor/modules.txt:github.com/opencontainers/runc/libcontainer/cgroups/ebpf/devicefilter
vendor/modules.txt:github.com/opencontainers/runc/libcontainer/cgroups/fs
vendor/modules.txt:github.com/opencontainers/runc/libcontainer/cgroups/fs2
vendor/modules.txt:github.com/opencontainers/runc/libcontainer/cgroups/fscommon
vendor/modules.txt:github.com/opencontainers/runc/libcontainer/cgroups/systemd
```

</details>

ehashman replied:
`go list -deps` for each cmd directory shows:

```
ehashman@red-dot:~/src/k8s$ for dir in cmd/*; do cd $dir; go list -deps | grep -q 'github.com/opencontainers/runc/libcontainer/cgroups/fs' && echo $dir; cd - &> /dev/null; done
cmd/genkubedocs
cmd/genman
cmd/kubelet
cmd/kubemark
```

with deps... Not sure how kubemark is pulling this in, will that be a problem?

liggitt replied:
kubemark runs a mock kubelet so basically has the same deps. If CI is passing, then I don't think it's a problem, but check with the scaleability team

kolyshkin replied:
I don't know if anything can function if there's no access to `/sys/fs/cgroup`...

But this is a valid point, and we can do better.

liggitt commented on vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs/fs.go:
```diff
@@ -1,165 +1,86 @@
+func init() {
+	// If using cgroups-hybrid mode then add a "" controller indicating
+	// it should join the cgroups v2.
+	if cgroups.IsCgroup2HybridMode() {
```
this is the call that can panic

mrunalp replied:
Good catch! will follow up on runc upstream.

mrunalp replied:
cc: @kolyshkin

mrunalp replied:
Opened https://github.com/opencontainers/runc/pull/3432.

liggitt commented:
noticed one unfortunate looking init() func addition that has a panic built in

dims commented:
@ehashman needs a rebase as well!

ehashman committed:
deps: update runc to 1.1.0

This updates vendored runc/libcontainer to 1.1.0,
and google/cadvisor to a version updated to runc 1.1.0
(google/cadvisor#3048).

Changes in vendor are generated by (roughly):

        ./hack/pin-dependency.sh github.com/google/cadvisor v0.44.0
        ./hack/pin-dependency.sh github.com/opencontainers/runc v1.1.0
        ./hack/update-vendor.sh
        ./hack/lint-dependencies.sh # And follow all its recommendations.
        ./hack/update-vendor.sh
        ./hack/update-internal-modules.sh
        ./hack/lint-dependencies.sh # Re-check everything again.

Co-Authored-By: Kir Kolyshkin <kolyshkin@gmail.com>

ehashman commented:
Rebased.

ehashman commented:
/test pull-kubernetes-node-e2e-containerd

ehashman commented:
Flake was https://github.com/kubernetes/kubernetes/issues/109082 - filed bug

liggitt commented:
/approve
for dependency mechanics, since linking to the cgroups package containing the init() panic possibility is limited to kubelet in production binaries... if at all possible, it would be good to pick up 1.1.1 with that bit dropped before release

/lgtm
based on @bobbypage @mrunalp lgtms above

mrunalp commented:
/lgtm

ehashman commented:
/retest-required

kolyshkin commented:
OK, runc 1.1.1 is now out, and this is a PR to replace this one (no changes other than rebase and use runc 1.1.1 instead of 1.1.0): https://github.com/kubernetes/kubernetes/pull/109091

rkojedzinszky commented:
Will it be backported to 1.22 and 1.23 branches?
